uitools/toolkitcpp.cmake

```diff
@@ -0,0 +1,88 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set(CPPPATH ${CMAKE_CURRENT_LIST_DIR}/cpp/Esri/ArcGISRuntime/Toolkit)
+
+set(REGISTERPATH ${CMAKE_CURRENT_LIST_DIR}/register/Esri/ArcGISRuntime/Toolkit)
+
+set(TOOLKITCPP_SOURCES
+    ${CPPPATH}/AuthenticationController.cpp
+    ${CPPPATH}/BasemapGalleryController.cpp
+    ${CPPPATH}/BasemapGalleryItem.cpp
+    ${CPPPATH}/BookmarksViewController.cpp
+    ${CPPPATH}/BookmarkListItem.cpp
+    ${CPPPATH}/CoordinateConversionConstants.cpp
+    ${CPPPATH}/CoordinateConversionController.cpp
+    ${CPPPATH}/CoordinateConversionOption.cpp
+    ${CPPPATH}/CoordinateConversionResult.cpp
+    ${CPPPATH}/CoordinateOptionDefaults.cpp
+    ${CPPPATH}/FloorFilterController.cpp
+    ${CPPPATH}/FloorFilterFacilityItem.cpp
+    ${CPPPATH}/FloorFilterLevelItem.cpp
+    ${CPPPATH}/FloorFilterSiteItem.cpp
+    ${CPPPATH}/LocatorSearchSource.cpp
+    ${CPPPATH}/Internal/BasemapGalleryImageProvider.cpp
+    ${CPPPATH}/Internal/GenericListModel.cpp
+    ${CPPPATH}/Internal/GenericTableProxyModel.cpp
+    ${CPPPATH}/Internal/MetaElement.cpp
+    ${CPPPATH}/NorthArrowController.cpp
+    ${CPPPATH}/OverviewMapController.cpp
+    ${CPPPATH}/PopupViewController.cpp
+    ${CPPPATH}/ScalebarController.cpp
+    ${CPPPATH}/SearchResult.cpp
+    ${CPPPATH}/SearchSourceInterface.cpp
+    ${CPPPATH}/SearchSuggestion.cpp
+    ${CPPPATH}/SearchViewController.cpp
+    ${CPPPATH}/SmartLocatorSearchSource.cpp
+    ${CPPPATH}/TimeSliderController.cpp
+    ${CPPPATH}/UtilityNetworkFunctionTraceResult.cpp
+    ${CPPPATH}/UtilityNetworkFunctionTraceResultsModel.cpp
+    ${CPPPATH}/UtilityNetworkListItem.cpp
+    ${CPPPATH}/UtilityNetworkTraceController.cpp
+    ${CPPPATH}/UtilityNetworkTraceStartingPoint.cpp
+    ${CPPPATH}/UtilityNetworkTraceStartingPointsModel.cpp
+    ${REGISTERPATH}/register.cpp
+    ${REGISTERPATH}/internal/register_cpp.cpp)
+
+set(TOOLKITCPP_RESOURCES
+    ${CMAKE_CURRENT_LIST_DIR}/images/esri_arcgisruntime_toolkit_images.qrc
+    ${CMAKE_CURRENT_LIST_DIR}/import/Esri/ArcGISRuntime/Toolkit/esri_arcgisruntime_toolkit_view.qrc)
+
+target_include_directories(${PROJECT_NAME} PRIVATE
+    ${CMAKE_CURRENT_LIST_DIR}/cpp
+    ${CPPPATH}
+    ${CMAKE_CURRENT_LIST_DIR}/register
+    ${REGISTERPATH})
+
+target_sources(${PROJECT_NAME} PRIVATE
+    ${TOOLKITCPP_SOURCES}
+    ${TOOLKITCPP_RESOURCES})
+
+find_package(Qt6 COMPONENTS REQUIRED QuickControls2 WebView Svg QuickLayouts)
+
+target_link_libraries(${PROJECT_NAME} PRIVATE
+    Qt6::QuickControls2
+    Qt6::WebView
+    Qt6::Svg)
+
+if(IOS)
+    target_link_libraries(${PROJECT_NAME} PRIVATE Qt6::qquicklayoutsplugin)
+else()
+    target_link_libraries(${PROJECT_NAME} PRIVATE Qt6::QuickLayouts)
+endif()
```

Comment:
Can you help me understand why this is needed at all? If we are using QuickLayouts, shouldn't it be added to `target_link_libraries` above?

Reply:
Good question, I haven't been able to get to the bottom of this yet.

This is what the robot tells me:

> The Qt6::qquicklayoutsplugin is a static version of the Qt6::QuickLayouts library. It's used on iOS because of the static linking requirement. On other platforms, the dynamic Qt6::QuickLayouts library is used instead.

And this is what Qt has to say (https://doc.qt.io/qt-6/porting-to-ios.html)
> In Qt for iOS, everything is compiled statically and placed into the application bundle. The applications are "sandboxed" inside their bundles and cannot make use of shared object files. Because of this, also the plugins used by the Qt modules need to be statically linked.

But that doesn't really answer why this is the only library that needs to be handled this way.

Reply:
I think it's ok that we keep it for now, but it feels like the kinda thing that will come back to haunt us at some point with future Qt versions.